### PR TITLE
added code for bresenham circlo algo in c & c++

### LINKED ISSUE #792: Breseham circle drawing algo
It is used to draw circle

### PR DESCRIPTION
**Pull request by tusharsabhani**
Fixes #792 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Code for bresenham circle drawing algo

#### Languages Used:
- C
- C++

#### Files Added:
- bresenham_circle_algo.c
- bresenham_circle_algo.cpp

### CODE REVIEW

**Comment by marwahmanbir** [comment]
Hey! Please complete the checklist above.

**Comment by tusharsabhani** [comment]
Done

**Comment by marwahmanbir** [comment]
A little edit in my previous comment: you have to complete the whole PR template and not only the checklist.
Your code looks good though.

**Comment by marwahmanbir** [comment]
Also, you can mark the checkbox like this (remove spaces):
[ ] unchecked
[ x ] checked

**Comment by tusharsabhani** [comment]
Done
Please have a look if anything else needs to be done

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#include<graphics.h>
+#include<stdlib.h>
+#include<stdio.h>
+#include<conio.h>
+void main()
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Add single newline before main  function.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+{
+    clrscr();
+
+    //r is the radius of the circle
+    //x0 and y0 is the centre of the circle
+    //x and y are an arbitrary point on circle
+    //p is the decision parameter
+    int x,y,x0,y0,p,r;
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Add a space after every comma.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    int gdriver = DETECT, gmode;
+    initgraph(&gdriver,&gmode, "");
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Add space before `&gmode`

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    printf("Enter the radius of circle.");
+    scanf("%d",&r);
+    printf("Enter co-ordinates of centre of circle.");
+    scanf("%d%d",&x0,&y0);
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Take input in new line.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    x=0;
+    y=r;
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Add single space before and after every operator.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    //putpixel is used to plot the point
+    putpixel(x0+x,y0-y,10);
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Add space after commas for better readability.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 7]
```diff
@@ -0,0 +1,55 @@
+#include<graphics.h>
+#include<stdlib.h>
+#include<stdio.h>
+#include<conio.h>
```
Comment (jainaman224) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Remove conio.h
It is not supported by standard gcc compiler.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.cpp [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+#include<iostream.h>
+void main()
+{
+    clrscr();
+
+    //r is the radius of the circle
+    //x0 and y0 is the centre of the circle
+    //x and y are an arbitrary point on circle
+    //p is the decision parameter
+    int x,y,x0,y0,p,r;
```
Comment (jainaman224) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.cpp]: Please use descriptive variable names e.g. `radius` for `r`.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.cpp [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+            {p=p+4*x+6;}
+        else
+            {p=p+4*(x-y)+10;
+             y--;}
+        x++;
+
+        //since circle is 8 point symmetric so using this property to plot the circle
+        //first and second input of putpixel are the coordiantes while the third one
+	//is the color of the line
+        putpixel(x0+x,y0+y,10);
```
Comment (jainaman224) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.cpp]: Please have a look at https://www.codesdope.com/cpp-gear-up/.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    //putpixel is used to plot the point
+    putpixel(x0+x,y0-y,10);
+
+    p=3-(2*r);
+    while(x<=y)
+   {
+        if(p<0)
+            {p=p+4*x+6;}
+        else
+            {p=p+4*(x-y)+10;
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Keep the parenthesis of each block in separate line just like you did for the opening parenthesis of while loop.
All binary operators except ` . ` should be separated from their operands by spaces. Blank spaces should never separate unary operators such as unary minus, increment ("++"), and decrement ("--") from their operands.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+             y--;}
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Similarly, put the closing parenthesis in separate line.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        x++;
+   
+
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Keep a single blank line before comments. Remove the extra one.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+        //since circle is 8 point symmetric so using this property to plot the circle
+	//first and second input of putpixel are the coordiantes while the third one
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: 4 space indentation.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+	//is the color of the line
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Fix indentation.

File: Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+        putpixel(x0+x,y0+y,10);
+        putpixel(x0-x,y0-y,9);
+        putpixel(x0+x,y0-y,8);
+        putpixel(x0-x,y0+y,7);
+        putpixel(x0+y,y0+x,6);
+        putpixel(x0-y,y0-x,5);
+        putpixel(x0+y,y0-x,4);
+        putpixel(x0-y,y0+x,3);
```
Comment (marwahmanbir) [inline Bresenham_Circle_Drawing_Algo/bresenham_circle_algo.c]: Single space after commas.

**Comment by tusharsabhani** [comment]
fixed all the indentation and spacing
please have a look

**Comment by marwahmanbir** [comment]
@tusharsabhani Good work! :+1:

**Comment by jainaman224** [comment]
@marwahmanbir Is it done?

**Comment by marwahmanbir** [comment]
@jainaman224 Yes

**Pull request merged**